### PR TITLE
Feature/refactor frontend#137

### PR DESCRIPTION
**Pull request by wasabideveloper**
Implementation of refactoring #137.

* ran checks
* ran tests
* ran flake8
* tested views by hand:
        /
        /orders
        /orders/{orderId}
        /orders/new [get/post]
        /downloads

@das-g, @njordan-hsr please review.

### CODE REVIEW

File: osmaxx-py/osmaxx/excerptexport/templates/excerptexport/partials/login_logout_link.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+{% if user.is_authenticated %}
+<a href="{% url 'excerptexport:logout' %}?next={% url 'excerptexport:index' %}">
+    <span class="menu_icon">&prod;&rarr;</span>&nbsp;{% trans 'Logout' %} <strong>{{ user.username }}</strong>
+</a>
+{% else %}
+<a href="{% url 'excerptexport:login' %}?next={% url 'excerptexport:index' %}">
+    <span class="menu_icon">&prod;&larr;</span>&nbsp;{% trans 'Login' %}
+</a>
+{% endif %}
```
Comment (njordan-hsr) [inline osmaxx-py/osmaxx/excerptexport/templates/excerptexport/partials/login_logout_link.html]: Can we get rid of the "No Newline at end of file"?

File: osmaxx-py/osmaxx/excerptexport/templates/excerptexport/partials/navigation.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        <li role="presentation" {% if view = 'downloads' %}class="active"{% endif %}>
+            <a href="{% url 'excerptexport:downloads' %}">
+                <span class="menu_icon">&#8681;</span>&nbsp;{% trans 'My downloads' %}
+            </a>
+        </li>
+        <li>
+            {% include 'excerptexport/partials/login_logout_link.html' with user=user %}
+        </li>
+    </ul>
+</nav>
```
Comment (njordan-hsr) [inline osmaxx-py/osmaxx/excerptexport/templates/excerptexport/partials/navigation.html]: Can we get rid of the "No Newline at end of file"?

Comment (wasabideveloper) [inline osmaxx-py/osmaxx/excerptexport/templates/excerptexport/partials/navigation.html]: Fixed and tried again to configure my IDE...

**Comment by njordan-hsr** [comment]
We could use [regroup](https://docs.djangoproject.com/en/1.8/ref/templates/builtins/#regroup) and a single query to generate the list, should I file an issue for this?

**Comment by wasabideveloper** [comment]
@njordan-hsr Which list?

**Comment by njordan-hsr** [comment]
@wasabideveloper Discussing it on #145 (phrased it a bit complicated above, it should have said 'for the optgroup listing').

**Pull request merged**